tests/Integration/StructuredData/SinglePostTest.php

```diff
@@ -430,6 +430,191 @@ public function test_http_canonicals_for_single_post(): void {
 		self::assertSame( 'https', $url['scheme'] );
 	}
 
+	/**
+	 * Check post modified date in Parsely metadata.
+	 *
+	 * @covers \Parsely\Parsely::construct_parsely_metadata
+	 * @uses \Parsely\Parsely::get_author_name
+	 * @uses \Parsely\Parsely::get_author_names
+	 * @uses \Parsely\Parsely::get_bottom_level_term
+	 * @uses \Parsely\Parsely::get_category_name
+	 * @uses \Parsely\Parsely::get_clean_parsely_page_value
+	 * @uses \Parsely\Parsely::get_coauthor_names
+	 * @uses \Parsely\Parsely::get_current_url
+	 * @uses \Parsely\Parsely::get_first_image
+	 * @uses \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::get_tags
+	 * @uses \Parsely\Parsely::post_has_trackable_status
+	 * @uses \Parsely\Parsely::update_metadata_endpoint
+	 * @group metadata
+	 */
+	public function test_metadata_post_modified_date(): void {
+		// Setup Parsely object.
+		$parsely         = new Parsely();
+		$parsely_options = get_option( Parsely::OPTIONS_KEY );
+
+		// Create a post with a date in the past.
+		$time_format      = 'Y-m-d\TH:i:s\Z';
+		$time_yesterday   = time() - DAY_IN_SECONDS;
+		$date_created     = gmdate( $time_format, $time_yesterday );
+		$date_created_gmt = gmdate( $time_format, ( $time_yesterday + get_option( 'gmt_offset' ) * HOUR_IN_SECONDS ) );
+		$post_id          = self::factory()->post->create(
+			array(
+				'post_date'     => $date_created,
+				'post_date_gmt' => $date_created_gmt,
+			)
+		);
+
+		// In the metadata, check that the post's created date is correct and
+		// that the last modified date is identical.
+		$post     = get_post( $post_id );
+		$metadata = $parsely->construct_parsely_metadata( $parsely_options, $post );
+		self::assertSame( $date_created, $metadata['dateCreated'] );
+		self::assertSame( $date_created, $metadata['dateModified'] );
+
+		// Update the post and reload metadata.
+		wp_update_post( array( 'ID' => $post_id ) );
+		$post_updated     = get_post( $post_id );
+		$metadata_updated = $parsely->construct_parsely_metadata( $parsely_options, $post_updated );
+
+		// In the metadata, check that the last modified date has been updated.
+		self::assertSame( $date_created, $metadata_updated['dateCreated'] );
+		self::assertNotSame( $date_created, $metadata_updated['dateModified'] );
```

Comment:
I have no objections about `test_metadata_post_modified_date()` being removed if reviewers think it was rendered redundant.

My opinion (and rationale behind this implementation) is that in all integration tests we should use WP functions as much as possible, since in a regular workflow the data is manipulated by WP functions, and as such we should test with the data received by them. So I don't find "modified object" tests bad, but testing with WP built-in functions is a "reality check" that should be done IMO.

This doesn't rule out modifying objects with extreme or unexpected values to see what happens. However, the premise is that WP functions are upstream and should have their own tests to verify their quality and results. If their tests are lacking, they should be added there.

So regarding the missed assertion:

- I do expect `wp_update_post()` to add a correct date (in format and value) to the database.
- As I could not hardcode a time using `wp_update_post()`, the time could (at least theoretically) change at the time of comparison. But something that could be done to improve this is compare that modified time is in a 1-2 second window from the current time. But even this could fail under certain circumstances (for example performance problems).
- Given the previous point, I decided just to test for `NotSame` and that this would be good enough.

Having said that, team feedback is more important than my own opinion, so if any adjustments must be made or the test must be removed, that's fine by me.

Reply:
> in all integration tests we should use WP functions as much as possible, since in a regular workflow the data is manipulated by WP functions, and as such we should test with the data received by them

I agree in general. If `set_metadata_post_times` (& by proxy `construct_parsely_metadata`) were operating against the "current" post or some global, I'd probably be in favor of going the extra mile to call `wp_update_post`, but since they just take a post object, I'm less pressed about it, personally.

Maybe we leave both and move the tests I added to `tests/Unit`...?  Your call.

Reply:
> Maybe we leave both and move the tests I added to `tests/Unit`...? 

By definition of the tests including `$this->factory->post->create();`, they can't be unit tests. You'd have to instantiate `WP_Post` directly, and I suspect that might be painful to populate without a DB entry to pull from.

Reply:
I'm not very clear on what needs to be done here. I suggest leaving all tests as they are (no move to unit tests) unless something needs to be removed. If we start adding lots of tests for metadata perhaps it would be good to split the file into many parts (?) - though splitting the file is not in the scope of this PR.

Reply:
> I suggest leaving all tests as they are

Works for me